simcon_project/users/views/create_assignment.py

```diff
@@ -0,0 +1,156 @@
+from users.views.researcher_home import is_researcher
+import json
+import datetime
+
+
+class CreateAssignmentView(TemplateView):
+    def get(self, request):
+        curResearcher=request.user
+
+        stud=Student.objects.all().filter(added_by=curResearcher).values('email', 'first_name','last_name','is_active')
+        label=SubjectLabel.objects.all().filter(researcher=curResearcher)
+        template=ConversationTemplate.objects.all().filter(researcher=curResearcher)
+
+        student=json.dumps(list(stud))
+        subLabel = serializers.serialize("json",label)
+        template = serializers.serialize("json",template)
+
+        return render(request, 'create_assignment.html', {
+            'student': student,
+            'subLabel': subLabel,
+            'template': template,
+        })
+
+#Transfer string type list to list type
+def decode(str):
+    if str[0]!='[':
+        print("Decode failed. The data is not a list type.")
+        return
+    temp=str[1:-1]
+    temp=temp.split(',')
+    listTmp=[]
+    for t in temp:
+        t=t[1:-1]
+        listTmp.append(t)
+    return listTmp
+
+def sendMail(subject, msg, recipient, researcher):
+    send_mail(subject,msg,researcher,recipient,fail_silently=False)
+
+#Determine if data is empty.
+def isNull(data):
+    if len(data)==0 or (len(data)==1 and data[0]==''):
+        return True
+    return False
+
+@user_passes_test(is_researcher)
+def add_data(request):
+    #Error signs and error messages.
+    errMsg=''
+    success=0
+
+    #Get raw data
+    data=request.POST
+    name=data.get('name')
+    date=data.get('date')
+    researcher=request.user
+    #researcher='researcher@111.com'
+    students=data.get('stuData')
+    templates=data.get('tempData')
+    labels=data.get('labelData')
+
+    #Transfer string type list to list type
+    students=decode(students)
+    templates=decode(templates)
+    labels=decode(labels)
+
+    #Verify date
+    today=datetime.datetime.combine(datetime.date.today(), datetime.time(0, 0))
+    dateTmp=datetime.datetime.strptime(date, "%Y-%m-%d")
+    assignToday=False
+    if dateTmp==today:
+        assignToday=True
+    if dateTmp<today:
+        success=1
+        errMsg=errMsg+'The assigned date cannot before today.\n\n'
+
+    #Save the data that can be directly assigned to assignment.
+    assignment=Assignment()
+    assignment.name=name
+    assignment.date_assigned=date
+    researcher=Researcher.objects.get(email=researcher)
+    assignment.researcher=researcher
+    assignment.save()
+
+    #Determine if students, labels, and templates are empty.
+    stuIsNull=isNull(students)
+    labelIsNull=isNull(labels)
+    tempIsNull=isNull(templates)
+
+    #Assign student information to assignment,
+    #and judge whether there is an error that both students and labels are empty.
+    if stuIsNull:
+        if labelIsNull:
+            success=1
+            errMsg=errMsg+'One of the students or labels must not be empty.\n\n'
```

Comment:
You must have at least one student or label added

Reply:
The main purpose of this step is to determine whether the student data should be written into the assignment. The content of the error message is that both cannot be empty, so the order of judgment does not affect the result. If the student is empty and the label is not empty, the data of the label will be added to the assignment when it runs to the part responsible for adding the label.